cli/lock.ml

```diff
@@ -137,6 +137,12 @@ let lockfile_path ~explicit_lockfile ~target_packages repo =
       Project.lockfile
         ~target_packages:(OpamPackage.Name.Set.elements target_packages)
         repo
+      |> Result.map_error ~f:(function `Msg msg ->
+             Rresult.R.msgf
+               "Could not infer the target lockfile name: %s\n\
+                Try setting it explicitly using --lockfile or add a project \
```

Comment:
I'm wondering whether it wouldn't be nice to use some `Fmt.styled` option for `--lockfile` to mark it up as a shell command. It is a small thing. What do you think?

Reply:
Yup definitely make sense to improve this, I'll give it a spin. I'm not sure how well that works with the intermediate string allocated.

Might actually be a better idea to log this ourselves and `exit 1` here rather than embeding color and style encoding into a string that's carried around.

Reply:
Come to think about it, you're probably right. This is more a `panic` (to use Rust lingo) level output from which recovery is impossible, carrying around a colorful string does not give much value.

OTOH, terminating the program from a random place feels a bit dirty. I wonder whether it wouldn't make sense to make this a polymorphic variant like ``` `Unrecoverable of string```. Thought that's probably overcomplicating it.

Reply:
The formatting cannot be embedded in the string, I just tried it.

I agree terminating the program isn't great. One thing worth considering is that we could log the error there and let the main function termintate it. We could indeed use a special variant for this so that we're sure it's not intercepted. One that would have no attached message could be nice as well since we can assume that the error was reported already. If the program is to be terminated there is no reason to carry around the error message anymore.

How does that sound?

Reply:
After giving this a quick spin, it's not as easy to deal with so let's leave it out of this PR.

It would be nice to have a proper way to deal with this but it requires more in depth changes and those would be better done on their own!